vermouth/processors/do_links.py

```diff
@@ -216,10 +220,24 @@ def match_link(molecule, link):
     GM = nx.isomorphism.GraphMatcher(molecule, link, node_match=_atoms_match)
 
     raw_matches = GM.subgraph_isomorphisms_iter()
+    avoid_doubles = []
     for raw_match in raw_matches:
         # raw_match: mol -> link
         # rev_raw_match: link -> mol
         rev_raw_match = {value: key for key, value in raw_match.items()}
+        
+        # When the 'other' operator (*) is used, links involving
+        # same atoms from different residues will be doubled over
+        # permutations. This needs to be avoided.
+        unique = {a.lstrip('*') if isinstance(a, str) else a for a in raw_match.values()}
+        if len(unique) == 1:
+            group = (unique.pop(), set(raw_match.keys()))
+            resnames = {molecule.nodes[idx]['resname'] for idx in group[1]}
```

Comment:
Why only resname and not chain? Or other attributes?

Reply:
It can only happen if the chains are _different_ (using *). However, it doesn't need only apply to same residue names. In practice it does now (CYS-CYS), and it _is_ a symmetry issue (the same pattern applies both ways). But there could be cases where symmetry in atom names exists withouth symmetry in residue names, so this part of the check should probably be removed.